docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+    Search functionality will be implemented at a later date.
+
+.. code-block:: python
+
+    routes = gateway_client.search_routes()
+    for route in routes:
+        print(route)
+
+4. Querying a particular route: ``query(route: str, data: Dict[str, Any])``
+
+The ``query`` method submits a query to a configured provider route.
+The data structure you send in the query depends on the route.
+
+Here are examples for the "completions", "chat", and "embeddings" routes:
+
+* ``Completions``
+
+.. code-block:: python
+
+    response = gateway_client.query("completions", {"prompt": "It's one small step for"})
+    print(response)
+
+* ``Chat``
+
+.. code-block:: python
+
+    response = gateway_client.query(
+        "chat", {"messages": [{"role": "user", "content": "Tell me a joke about rabbits"}]}
+    )
+    print(response)
+
+* ``Embeddings``
+
+.. code-block:: python
+
+    response = gateway_client.query(
+        "embeddings", {"texts": ["It was the best of times", "It was the worst of times"]}
+    )
+    print(response)
+
+Further route types will be added in the future.
+
+These examples cover the public methods in the MlflowGatewayClient class, each demonstrating the method's function and usage.
+
+Fluent API
+----------
+The ``fluent`` API is designed to provide a user-friendly interface for interacting with the `MLflow Gateway`.
+It's a higher-level abstraction over the ``MlflowGatewayClient``, simplifying common operations and interactions with the Gateway.
+
+The ``fluent`` API includes functions such as ``get_route()``, ``search_routes()``, and ``query()``, e
+ach providing a simple, straightforward way to interact with the Gateway.
+
+* ``get_route(name: str) -> Route``: This function allows users to retrieve a specific ``route`` configuration from the `MLflow Gateway` service.
+
+* ``search_routes(search_filter: Optional[str] = None) -> List[Route]``: This function provides the ability to fetch all the routes from the Gateway service. The search filter is not functional currently and the function returns all the routes irrespective of the filter. If an argument is passed, an exception will be thrown to warn of this current lack of functionality.
+
+* ``query(route: str, data)``: This function makes it easy to send a ``query`` to a configured service through a named route on the Gateway Server. It handles the specifics of interfacing with the underlying ``MlflowGatewayClient`` and the configured ``route``, providing a simpler way to issue requests and get responses.
+
+With the ``fluent`` API, working with the `MLflow Gateway` is as easy as making a function call.
+It provides a cleaner, more Pythonic way of interacting with the Gateway service, freeing you to
+focus on what's really important - getting the results you need from model services.
+
+To use the ``fluent`` API, here are some examples:
```

Comment:
Maybe "For the ``fluent`` API, here are some examples:" ? "To use..." doesn't quite parallel "here are..."